packages/template-retail-react-app/app/hooks/use-shipment-operations.js

```diff
@@ -0,0 +1,174 @@
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {useShopperBasketsMutation} from '@salesforce/commerce-sdk-react'
+import {useCallback} from 'react'
+import {useErrorHandler} from '@salesforce/retail-react-app/app/hooks/use-error-handler'
+import {cleanAddressForOrder} from '@salesforce/retail-react-app/app/utils/shipment-utils'
+
+/**
+ * Hook for basic shipment CRUD operations
+ * Focused only on shipment management (API calls)
+ * @param {string} basketId - The basket ID
+ * @returns {Object} Object containing shipment operation functions
+ */
+export const useShipmentOperations = (basketId) => {
+    const handleError = useErrorHandler()
+    const createShipmentMutation = useShopperBasketsMutation('createShipmentForBasket')
+    const removeShipmentMutation = useShopperBasketsMutation('removeShipmentFromBasket')
+    const updateShipmentMutation = useShopperBasketsMutation('updateShipmentForBasket')
+    const updateShippingMethodMutation = useShopperBasketsMutation(
+        'updateShippingMethodForShipment'
+    )
+
+    /**
+     * Creates a new shipment
+     * @param {Object} address - The shipping address for the shipment
+     * @param {Object} options - Additional options for shipment creation
+     * @param {string} options.shippingMethodId - The shipping method ID
+     * @param {string} options.storeId - The store ID for pickup shipments
+     * @returns {Promise<string>} The created shipment ID
+     */
+    const createShipment = useCallback(
+        async (address, options = {}) => {
+            if (!basketId) {
+                throw new Error('Missing basketId')
+            }
+
+            try {
+                const body = {}
+
+                if (address) {
+                    body.shippingAddress = cleanAddressForOrder(address)
+                }
+
+                if (options.shippingMethodId) {
+                    body.shippingMethod = {
+                        id: options.shippingMethodId
+                    }
+                }
+
+                if (options.storeId) {
+                    body.c_fromStoreId = options.storeId
+                }
+
+                const response = await createShipmentMutation.mutateAsync({
+                    parameters: {
+                        basketId
+                    },
+                    body
+                })
+
+                // Find the newly created shipment
+                const newShipment = response?.shipments?.find(
+                    (shipment) => !response.shipments.slice(0, -1).includes(shipment)
+                )
+
+                return newShipment?.shipmentId
+            } catch (error) {
+                handleError('Failed to create shipment', error)
+                throw error
+            }
+        },
+        [basketId, createShipmentMutation, handleError]
```

Comment:
Same as above, do we really need `createShipmentMutation, handleError` as dependencies? How would that work?